src/utils/url.ts

```diff
@@ -0,0 +1,70 @@
+import { NotSupportedPlatformError } from "../main/exceptions/not-supported-platform";
+
+const MACOS_FILE_NAME = "macOS.tar.gz";
+const LINUX_FILE_NAME = "Linux.tar.gz";
+const WINDOWS_FILE_NAME = "Windows.zip";
+
+export interface IDownloadUrls {
+  launcher: string;
+  player: string;
+}
+
+export function getDownloadUrls(
+  baseUrl: string,
+  netenv: string,
+  apvVersion: number,
+  platform: NodeJS.Platform
+): IDownloadUrls {
+  // TODO: fix project version number hard coding: 1.
+  const launcherUrl = buildDownloadUrl(
+    baseUrl,
+    netenv,
+    apvVersion,
+    "launcher",
+    1,
+    platform
+  );
+  const playerUrl = buildDownloadUrl(
+    baseUrl,
+    netenv,
+    apvVersion,
+    "player",
+    1,
+    platform
+  );
+
+  return {
+    launcher: launcherUrl,
+    player: playerUrl,
+  };
+}
+
+export function buildDownloadUrl(
+  baseUrl: string,
+  env: string,
+  rc: number,
+  project: "player" | "launcher",
+  projectVersion: number,
+  platform: NodeJS.Platform
+): string {
+  const fn = BINARY_FILENAME_MAP[platform];
```

Comment:
```suggestion
  const filename = BINARY_FILENAME_MAP[platform];
```

`fn` is commonly used as an alias for function, confusing this code.

Reply:
Done